### PR TITLE
Propagate quarkus OPTS to openshift deployment

### LINKED ISSUE #1034: Openshift build deployment strategy does not propagate QUARKUS_OPTS
Using OpenShiftDeploymentStrategy.Build in `@OpenShiftScenario` ignores quarkus-opts properties from properties file, like:
`quarkus.openshift.env.vars.quarkus-opts`
Deployment in openshift has always empty QUARKUS_OPTS as set in https://github.com/quarkus-qe/quarkus-test-framework/blob/main/quarkus-test-openshift/src/main/resources/quarkus-build-openshift-template.yml#L94

These options should be propagated to the openshift deployment.

### PR DESCRIPTION
**Pull request by mocenas**
### Summary

Fix #1034 
Propagate quarkus OPTS to openshift deployment.

Please check the relevant options

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Dependency update
- [ ] Refactoring
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a documentation update
- [X] This change requires execution against OCP (use `run tests` phrase in comment)

### Checklist:
- [ ] Example scenarios has been updated / added
- [X] Methods and classes used in PR scenarios are meaningful
- [X] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

**Comment by mocenas** [comment]
run tests

**Review by fedinskiy — COMMENTED** [review]
Hmm, this is strange. 
1. application.properties is included in the build and so it should be used anyway. If it is not included, a lot more things would be affected.
2. Why do you need to pass `quarkus.openshift.env.vars.quarkus-opts` to a pod, which already on openshift? From  the name of it, it should be passed to the build made my quarkus-openshift extension, and not by us
3. The only thing, which mentions this property is that bug issue[1] created by us. Is it even exists?

[1] https://github.com/quarkusio/quarkus/issues/29642

**Comment by mocenas** [comment]
> Hmm, this is strange.
> 
>     1. application.properties is included in the build and so it should be used anyway. If it is not included, a lot more things would be affected.
> 
>     2. Why do you need to pass `quarkus.openshift.env.vars.quarkus-opts` to a pod, which already on openshift? From  the name of it, it should be passed to the build made my quarkus-openshift extension, and not by us
> 
>     3. The only thing, which mentions this property is that bug issue[1] created by us. Is it even exists?
> 
> 
> [1] [quarkusio/quarkus#29642](https://github.com/quarkusio/quarkus/issues/29642)

Yes QUARKUS_OPTS really does exists and is effective on native. Where `quarkus.openshift.env.vars.quarkus-opts`  is ineffective and needs to be passed to the pod this way. Basically `quarkus.openshift.env.vars.quarkus-opts` should be translated/transfered into QUARKUS_OPTS. 

See [Josef's investigation about the issue|https://github.com/quarkusio/quarkus/issues/29642#issuecomment-1355041068) about that.

There is a big difference if we deploy the build using openshift-extension, or using our own Build. (see [deployment strategies](https://github.com/quarkus-qe/quarkus-test-framework/blob/main/quarkus-test-openshift/src/main/java/io/quarkus/test/scenarios/OpenShiftDeploymentStrategy.java). If we use openshift-extension then `quarkus.openshift.env.vars.quarkus-opts` is handled by it and it is used to set QUARKUS_OPTS.

But using `OpenShiftDeploymentStrategy.Build` (default)  makes it to be deployed to openshift via test-framework, where QUARKUS_OPTS is handled manually by us. And prior to this PR, it was always set to empty (see changes in template).

**Comment by mocenas** [comment]
Openshift failure is unrelated, executor failed mid tests. I will rerun the tests.

**Comment by mocenas** [comment]
run tests

File: quarkus-test-openshift/src/main/java/io/quarkus/test/services/quarkus/BuildOpenShiftQuarkusApplicationManagedResource.java [hunk 1]
```diff
@@ -137,4 +144,15 @@ private String getS2iImage() {
         return model.getContext().getOwner().getProperty(s2iImageProperty.getPropertyKey())
                 .orElseGet(() -> s2iImageProperty.get(model.getContext()));
     }
+
+    private String getQuarkusOpts() {
+        Path applicationProperties = model.getComputedApplicationProperties();
+        if (Files.exists(applicationProperties)) {
+            Map<String, String> properties = PropertiesUtils.toMap(applicationProperties);
+            if (properties.containsKey(QUARKUS_OPENSHIFT_OPTS_PROPERTY)) {
```
Comment (fedinskiy) [inline quarkus-test-openshift/src/main/java/io/quarkus/test/services/quarkus/BuildOpenShiftQuarkusApplicationManagedResource.java:152]: Maybe getting the value and then checking for `null` will be more clear.

**Review by fedinskiy — CHANGES_REQUESTED** [review]
This action is a little bit surprising from user's standpoint (since we access property, used by `quarkus-openshift`, while not using the extension).

I am not sure we should do it, but we definitely need to log this action (maybe even with content of `QUARKUS_OPTS` property)

File: quarkus-test-openshift/src/main/java/io/quarkus/test/services/quarkus/BuildOpenShiftQuarkusApplicationManagedResource.java (hunk 1 above)
Comment (mocenas) [reply quarkus-test-openshift/src/main/java/io/quarkus/test/services/quarkus/BuildOpenShiftQuarkusApplicationManagedResource.java:152]: That IMHO depends on programming style. For me, this is more clear.

**Review by fedinskiy — APPROVED** [review]
Thank you!

**Pull request merged**